pkg/client/fn_service_test.go

```diff
@@ -27,15 +27,19 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"github.com/funlessdev/fl-cli/test/mocks"
 	openapi "github.com/funlessdev/fl-client-sdk-go"
 )
 
 func TestFnInvoke(t *testing.T) {
-	testFn := "test-fn"
-	testNs := "test-ns"
+	testFn := "test_fn"
+	testNs := "test_ns"
 	var testArgs map[string]interface{} = map[string]interface{}{"name": "Some name"}
 
 	testCtx := context.Background()
+	mockValidator := mocks.NewInputValidatorHandler(t)
```

Comment:
Do we really need to mock the validator? It doesn't have any interaction with external systems. We can just use the concrete type and do the real validation.

Reply:
It was mostly done to use the `AssertNumberOfCalls` in tests, to ensure fn/mod functions call the input validation, while testing the actual validation in a separate file. Without the mock we would have to check the validation in each fn/mod function call I think

Reply:
Perhaps we could use the concrete type and have a test in mod/fn_service where we try to create a function with an invalid name and check that it returns an error and the request doesn't go through, so we avoid mocking and we check that the validator actually stops the request.

If you want to give it a try and see how the tests look like it'd be great. Afterwards we can either integrate that or just merge how it is now.

Reply:
After trying the test design using the concrete validator instead of the mock, I found that (to have a complete coverage), new tests should be added for all `fn` and `mod` subcommands. This would result in simply replicating the validator tests, instead of isolating them in one file. 
I think we can keep the tests as they are right now, and if the validator is changed or expanded in the future, simply update its tests; if, instead, we find that the validator is better suited as a simple auxiliary function, we can scrap the whole interface/struct and integrate the relative tests in the subcommands.